packages/evolution-backend/src/services/routing/RouteCalculationFromTransition.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2024, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+
+// This file contains route calculation utilities that can be used by the
+// survey. The should call chaire-lib's routing services to calculate the route.
+import { RoutingOrTransitMode } from 'chaire-lib-common/lib/config/routingModes';
+import { RouteCalculationParameter, RoutingTimeDistanceResultByMode } from './types';
+import projectConfig from '../../config/projectConfig';
+import { URL } from 'url';
+
+// A name for this calculation source
+const CALCULATION_SOURCE = 'transitionApi';
```

Comment:
We should have have as a type, so that we can use it as an Enum elsewhere.

Reply:
I'm not sure it should be an enum. Evolution may not know all possible sources (for example in od_longue distance, we implement a fallback when there is error or no routing).